workflows/flows/analyse_study_tasks/run_virify_pipeline_via_samplesheet.py

```diff
@@ -0,0 +1,201 @@
+
+    # Path to the GFF file
+    gff_dir = virify_outdir / EMG_CONFIG.virify_pipeline.final_gff_folder
+
+    # Check if the GFF directory exists
+    if not gff_dir.exists():
+        logger.warning(
+            f"GFF directory {gff_dir} does not exist. No GFF file to add to downloads."
+        )
+        return
+
+    # Find GFF files in the directory
+    gff_files = list(gff_dir.glob("*.gff"))
+
+    if not gff_files:
+        logger.warning(
+            f"No GFF files found in {gff_dir}. No GFF file to add to downloads."
+        )
+        return
+
+    # Use the first GFF file found
+    gff_file = gff_files[0]
+
+    # Create a relative path from the analysis results directory
+    rel_path = f"virify/{gff_file.name}"
+
+    # Create a download file object
+    download = DownloadFile(
+        path=rel_path,
+        alias=f"virify_{gff_file.name}",
+        download_type=DownloadType.GENOME_ANALYSIS,
+        file_type=DownloadFileType.GFF,
+        long_description="Viral genome annotation from VIRify pipeline",
+        short_description="Viral annotations",
+        download_group="virify",
+    )
+
+    # Add the download to the analysis
+    try:
+        analysis.add_download(download)
+        logger.info(
+            f"Added virify GFF file {gff_file.name} to analysis {analysis.accession} downloads"
+        )
+    except FileExistsError:
+        logger.warning(
+            f"Download with alias {download.alias} already exists for analysis {analysis.accession}"
+        )
+
+
+@flow(name="Run VIRIfy pipeline via samplesheet", log_prints=True)
+def run_virify_pipeline_via_samplesheet(
+    mgnify_study: analyses.models.Study,
+    assembly_analyses: List[analyses.models.Analysis],
+    assembly_pipeline_outdir: Path,
+):
+    """
+    Run the VIRIfy pipeline for a set of assemblies using the downstream samplesheet.
+
+    :param mgnify_study: The MGnify study
+    :param assembly_analyses: List of assembly analyses
+    :param assembly_pipeline_outdir: The output directory of the assembly pipeline
+    """
+    logger = get_run_logger()
+
+    # Path to the virify samplesheet
+    virify_samplesheet_path = (
+        assembly_pipeline_outdir
+        / EMG_CONFIG.assembly_analysis_pipeline.downstream_samplesheets_folder
+        / EMG_CONFIG.assembly_analysis_pipeline.virify_samplesheet
+    )
+
+    # Check if the virify samplesheet exists
+    if not virify_samplesheet_path.exists():
+        logger.warning(
+            f"Virify samplesheet {virify_samplesheet_path} does not exist. Skipping virify pipeline."
+        )
+        # TODO: update the status of the job
+        # for assembly_analysis in assembly_analyses:
+        #     assembly_analysis ..
+        return
+
+    # Mark analyses as started
+    # Add the MAP GFF file to each analysis's downloads
+    # TODO: I feel we need and intermediate status  here, otherwise the analysis will cycle through running/started a
+    # a few times
```

Comment:
I think that's okay – in that the statuses are multiple, not a single state. It (should) *always* have the status of STARTED if it was ever started. I think it's more that you want other statuses for the sub-pipelines starting/failing/finishing perhaps.

Reply:
This could also be implemented just with the `_reason` though